pkg/resource/aws/aws_security_group_rule.go

```diff
@@ -32,3 +40,72 @@ func (r *AwsSecurityGroupRule) TerraformType() string {
 func (r *AwsSecurityGroupRule) CtyValue() *cty.Value {
 	return r.CtyVal
 }
+
+func CreateSecurityGroupRuleIdHash(attrs *resource.Attributes) string {
+	var buf bytes.Buffer
+	buf.WriteString(fmt.Sprintf("%s-", *attrs.GetString("security_group_id")))
+	if attrs.GetInt("from_port") != nil && *attrs.GetInt("from_port") > 0 {
+		buf.WriteString(fmt.Sprintf("%d-", *attrs.GetInt("from_port")))
+	}
+	if attrs.GetInt("to_port") != nil && *attrs.GetInt("to_port") > 0 {
+		buf.WriteString(fmt.Sprintf("%d-", *attrs.GetInt("to_port")))
+	}
+	buf.WriteString(fmt.Sprintf("%s-", *attrs.GetString("protocol")))
+	buf.WriteString(fmt.Sprintf("%s-", *attrs.GetString("type")))
+
+	if attrs.GetStringSlice("cidr_blocks") != nil {
+		for _, v := range attrs.GetStringSlice("cidr_blocks") {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+	}
+
+	if attrs.GetStringSlice("ipv6_cidr_blocks") != nil {
+		for _, v := range attrs.GetStringSlice("ipv6_cidr_blocks") {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+	}
+
+	if attrs.GetStringSlice("prefix_list_ids") != nil {
+		for _, v := range attrs.GetStringSlice("prefix_list_ids") {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+	}
+
+	if (attrs.GetBool("self") != nil && *attrs.GetBool("self")) ||
+		(attrs.GetString("source_security_group_id") != nil && *attrs.GetString("source_security_group_id") != "") {
+		if attrs.GetBool("self") != nil && *attrs.GetBool("self") {
+			buf.WriteString(fmt.Sprintf("%s-", *attrs.GetString("security_group_id")))
+		} else {
+			buf.WriteString(fmt.Sprintf("%s-", *attrs.GetString("source_security_group_id")))
+		}
+		buf.WriteString("-")
+	}
+
+	return fmt.Sprintf("sgrule-%d", hashcode.String(buf.String()))
+}
+
+func initAwsSecurityGroupRuleMetaData(resourceSchemaRepository resource.SchemaRepositoryInterface) {
+	resourceSchemaRepository.SetNormalizeFunc(AwsDefaultSecurityGroupResourceType, func(val *resource.Attributes) {
+		val.SafeDelete([]string{"self"})
+		if sgid := val.GetString("security_group_id"); sgid != nil && *sgid == "" {
+			val.SafeDelete([]string{"security_group_id"})
+		}
+
+		// On first run, this field is set to null in state file and to "" after one refresh or apply
+		// This ensure that if we find a nil value we dont drift
+		if desc := val.GetString("description"); desc != nil && *desc == "" {
```

Comment:
Maybe you can use `val.DeleteIfDefault("description")` ? For `security_group_id` too